authentik/flows/stage.py

```diff
@@ -92,7 +92,11 @@ def get_response_instance(self, data: QueryDict) -> ChallengeResponse:
 
     def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         """Return a challenge for the frontend to solve"""
-        challenge = self._get_challenge(*args, **kwargs)
+        try:
+            challenge = self._get_challenge(*args, **kwargs)
+        except StageInvalidException as exc:
+            self.logger.debug("Got StageInvalidException", exc=exc)
+            return self.executor.stage_invalid()
```

Comment:
Perhaps this is not even needed? I believe `StageInvalidException` should only be thrown by a validator, in our case, the line below this.